Выполнил задание

mechmedved opened the pull request:
Выполнил задание 8 лекции: 
Вопрос: не понял как создать объекты Human father, Human mother в конструкторе Human

mechmedved committed:
Выполнил задание

RinatBeybutov commented:
Задание не поняли:
1. Нужно создать ОДИН класс Human, наследников и другие классы создавать не нужно.
2. Распечатывать в конструкторе не нужно, нужно переопределить метод toString
3. Нужно создать 2 конструктора для Human: один принимает только имя, пол и возраст, а второй имя, пол, возраст и два объекта Human, которые и будут родителями для этого человека. 
4. HumanHelper здесь не нужен

mechmedved committed:
Поправил домашку:
Убрал избыточное количество классов
Добавил объекты human(father) human(mother)

mechmedved commented:
Поправил домашку: - убрал избыточное количество классов ; - добавил human(father) human(mother)

RinatBeybutov commented on src/GrandParent.java:
```diff
@@ -0,0 +1,10 @@
+
+        super(name,age,sex);
+
+        System.out.println("Name: " + getName() + ", " +
```
удаление старых классов не произошло, их нужно удалить у себя, а потом запушить этот результат

RinatBeybutov commented on src/Main.java:
```diff
@@ -1,15 +1,35 @@
+        for (Human human : family) {
+            System.out.println("Name: " + human.getName());
+            System.out.println("Age: " + human.getAge());
+            System.out.println("Gender: " + human.isMale());
```
здесь должен быть вызван один метод toString у текущего human внутри цикла. Соответственно нужно этот метод переопределить в классе Human

RinatBeybutov commented on src/Human.java:
```diff
@@ -0,0 +1,50 @@
+        return name;
+    }
+
+    public String getMother() {
```
если метод называется getMother, то он по логике должен возвращать Human, то есть ссылку на сам объект. Здесь лучше назвать метод getMotherName. С учетом метода toString этот метод может быть избыточным

RinatBeybutov commented on src/Human.java:
```diff
@@ -0,0 +1,50 @@
+        return null;
+    }
+
+    public String getFather() {
```
то же самое про название

mechmedved committed:
Поправил getMoverName getFatherName

mechmedved commented:
У меня в репозитории осталось 2а класса Human и Main, запушил, предполагаю что они удаляются также в облаке
Переопределил метод toString в классе Human

RinatBeybutov commented on src/Human.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Override
+    public String toString() {
+        if(mother != null && father != null) {
```
здесь лучше всего сделать отдельно без этого if 
String motherName = mother == null ? "неизвестно" : mother.getName()
String fatherName = father == null ? "неизвестно" : father.getName()
return String.format("Name - %s, age - %d ...", name, age, motherName...)

mechmedved replied:
Спасибо , выглядит компактно )

mechmedved committed:
Поправил метод toString() в классе Human
